### ISSUE TITLE
Final Peer Review by ld457

### ISSUE THREAD

**Issue by ld457** [gregstepaniouk/MLBigDataProject#13]
The project aims at predicting the spread an investor would expect to pay when placing an order of a stock in NYSE. The input space includes features about markets conditions, security specific factors and properties about the order flow to the Exchange. The output space is the average effective spread. 

Things I like about the project:
1. The project includes a clear explanation for the data sets used for prediction. It can also use reasonable methods to deal with missing data and transform and encode features to overcome the flaws in the data.
2. It can choose appropriate loss functions for linear fitting based on the analysis of the output and input space. It can also use cross validation appropriately to find the optimal hyper-parameter for the models.
3. The part of Tree model can illustrate clearly the steps involved in the development of the models.
4. The report can give a great visualization so it is easy to understand data and the results of different models.

Areas for improvement:
1. Regression tree models are often used in classification problems, but the output space in this project is not classes. So, it is better to illustrate how the models transform the output variables into different classes. 
2. I think the reason the project choose the minimum leaf size in the tree is not strong enough. I would suggest to use cross validation to find out the best minimum leaf size according to the validation loss. 
3. It is better to explain how you separate the training and test set for the prediction.

**Comment by gregstepaniouk** [comment]
I appreciate your feedback but would like to point the following out:
1. **Regression tree** models are specifically used for a continuous output space. **classification trees** are used for classification.  [http://www.stat.wisc.edu/~loh/treeprogs/guide/wires11.pdf](url)

2. There is actually a graph of **10 fold cross validated MSE** with minimum leaf size and you can see that it keeps dropping so I chose the min leaf size that limits complexity at a point where I felt that qualitatively the MSE tradeoff was fair. I then go on to justify my decision by showing that a tree with smaller min leaf size overfit the training data. I agree that I could have cross-validated on the random forest and not only the single tree but I was limited by computing power at that point. There is actually a mention of this in the paper.

3. I agree that I neglected to mention the test/training split was 80/20.
